julia/src/model.jl

```diff
@@ -77,8 +80,9 @@ mutable struct StanModel
             ccall(
                 Libc.Libdl.dlsym(sm.lib, "bs_destruct"),
                 UInt32,
-                (Ptr{StanModelStruct},),
+                (Ptr{StanModelStruct}, Ref{Cstring}),
                 sm.stanmodel,
+                C_NULL,
```

Comment:
Why not "catch" and handle the error message similarly here, with `_handle_error(lib, err, "bs_descruct")`?

Reply:
Good question! Two reasons:

1. Julia has [rules about what you can do inside of a finalizer](https://docs.julialang.org/en/v1/base/base/#Base.finalizer). In particular, I/O is a no go, so I assume erroring is bad.
2. We never really provide useful information when `destruct` fails:
  https://github.com/roualdes/bridgestan/blob/6e50958cc5f2079fc332807619e9b6637e259f85/src/bridgestan.cpp#L36-L42
  In fact, I am pretty sure we can make `~bs_model()` a `noexcept` function. I believe it will never throw. So I might clean this up to avoid that entirely.

Reply:
Confirmed with @bob-carpenter that our destructor will not throw, so this is dead code. Going to clean up now